### PR TITLE
Add derivation for google ceres-solver library

### PR DESCRIPTION
**Pull request by giogadi**
This PR adds a derivation to development/libraries for Google's ceres-solver nonlinear optimization library for C++.

### CODE REVIEW

File: pkgs/development/libraries/ceres-solver/default.nix [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    -DEIGEN_INCLUDE_DIR=${eigen}/include/eigen3
+    ${if !useGlog then "-DMINIGLOG=ON" else ""}
+  ";
+
+  meta = with stdenv.lib; {
+    description = "C++ library for modeling and solving large, complicated optimization problems";
+    license = licenses.bsd3;
+    homepage = "http://ceres-solver.org";
+    maintainers = with stdenv.lib.maintainers; [ giogadi ];
+    inherit version;
```
Comment (peti) [inline pkgs/development/libraries/ceres-solver/default.nix]: Why do you inherit `version`here?

Comment (giogadi) [inline pkgs/development/libraries/ceres-solver/default.nix]: I'll admit that I used the derivation of [Eigen](https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/libraries/eigen/default.nix) as a model for this. Is including the version in the meta-attributes discouraged?

Comment (peti) [inline pkgs/development/libraries/ceres-solver/default.nix]: I wouldn't say that having `meta.version` discouraged per se, but adding attributes that aren't used anywhere is.

Comment (giogadi) [inline pkgs/development/libraries/ceres-solver/default.nix]: Sorry, I assumed the more information I could provide to the meta attributes, the better. Shall I open a PR to remove that attribute?

**Comment by peti** [comment]
No, it's not that important.

**Pull request merged**